src/main/java/duke/tasks/FileManager.java

```diff
@@ -0,0 +1,162 @@
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class FileManager {
+    public static final String FILE_PATH = "data/duke.txt";
+    private File taskFile;
+
+    public FileManager() {
+        taskFile = new File(FILE_PATH);
+    }
+
+    public void writeToDoToFile(Task task, int taskNumber, TaskManager taskManager) {
+        try {
+            appendToFile("T" + " | " + task.getStatusIcon() + " | " + task.getDescription());
+            if (taskNumber != taskManager.getNumberOfTasksAdded() - 1) {
+                appendToFile(System.lineSeparator());
+            }
+        } catch (IOException e) {
+            System.out.println("Oops, something went wrong! " + e.getMessage());
+        }
+    }
+
+    public void writeDeadlineToFile(Deadline deadline, int taskNumber, TaskManager taskManager) {
+        try {
+            appendToFile("D" + " | " + deadline.getStatusIcon() + " | "
+                    + deadline.getDescription() + " | " + deadline.getBy());
+            if (taskNumber != taskManager.getNumberOfTasksAdded() - 1) {
+                appendToFile(System.lineSeparator());
+            }
+        } catch (IOException e) {
+            System.out.println("Oops, something went wrong! " + e.getMessage());
+        }
+    }
+
+    public void writeEventToFile(Event event, int taskNumber, TaskManager taskManager) {
+        try {
+            appendToFile("E" + " | " + event.getStatusIcon() + " | "
+                    + event.getDescription() + " | " + event.getAt());
+            if (taskNumber != taskManager.getNumberOfTasksAdded() - 1) {
+                appendToFile(System.lineSeparator());
+            }
+        } catch (IOException e) {
+            System.out.println("Oops, something went wrong! " + e.getMessage());
+        }
+    }
+
+    /**
+     * Inputs the whole task file to the task list
+     *
+     * @param taskFile the task file to be read
+     * @param taskManager the task manager class to access add task methods
+     * @throws FileNotFoundException if the file is not found in the directory
+     */
+    public void retrieveFile(File taskFile, TaskManager taskManager) throws FileNotFoundException {
+        File f = taskFile;
+        Scanner s = new Scanner(f);
+        int taskNumber = 1;
+        while (s.hasNext()) {
+            String input = s.nextLine();
+            fillArrayFromFile(input, taskManager, taskNumber);
+            taskNumber++;
+        }
+    }
+
+    /**
+     * Inputs the task from the task file to the task list one by one
+     *
+     * @param input the task to be inputted to the task list
+     * @param taskManager the task manager class to access add task methods
+     * @param taskNumber the ith number of task from the top of the list
+     */
+    public void fillArrayFromFile(String input, TaskManager taskManager, int taskNumber) {
```

Comment:
This method name could be made more descriptive. Instead of `fillArrayFromFile` may be it can be `loadTasksFromFile`